2021/09/deploying-dotnet-5-app-iis.md

```diff
@@ -0,0 +1,74 @@
+---
+author: "Juan Pablo Ventoso"
+date: 2021-09-16
+title: "Deploying a .NET 5 app on IIS"
+tags:
+- dotnet
+- csharp
+- iis
+---
+
+![Puzzle](/blog/2021/09/deploying-dotnet-5-app-iis/puzzle-cropped.jpg)
+
+[Photo](https://flic.kr/p/4riBj8) by [Christian Cable](https://www.flickr.com/photos/nexus_icon/)
+
+.NET 5 is been around for a few years now, after being released at the [.NET Conf 2020](/blog/2020/11/dotnet-5-released-net-conf-2020/), containing the best of both worlds: .NET Core, including multi-platform support and several performance improvements, and .NET Framework, including Windows desktop development support with WPF and Windows Forms (UWP is also supported, but not officially yet).
+
+A .NET Core-based project can be published into any platform (as long as we’re not depending on libraries targeted to .NET Framework), allowing us to save costs by hosting in Linux servers and increasing performance for scalability. But most developers are still using Windows with Internet Informacion Services (IIS) as the publishing target: This might be related to the almost 20 years of history of .NET Framework, comparing to the relatively short history of .NET Core, launched in 2016.
+
+### Our .NET project
+
+We won’t review the steps needed to set up a new .NET 5 project, since this time we are only focusing on publishing what we already have developed. But to understand how our application will integrate with IIS and the framework, it’s important to note a fundamental change any .NET 5 project has in comparison with a .NET Framework one:
+
+Since .NET 5 is .NET Core in its foundation, our project output will actually be a console application. If we create a new .NET Core project, no matter which version we are using, we will find a `Program.cs` file in the root with an application entry point in it, that will look somewhat similar to the one below:
+
+![A common .NET Core app entry point](/blog/2021/09/deploying-dotnet-5-app-iis/dotnet-core-entry-point.jpg)
```

Comment:
Could you paste this program as text instead of an image? Keeps the page lighter, is searchable, copy/pasteable, etc.

Reply:
Done!